### PR TITLE
[IMPROVE] Change logic to generate token on Live Chat

### PR DESCRIPTION
**Pull request by cauefcr**
After reading the report on https://app.clickup.com/t/1nnnqte i suggest improving the token generation code so that even if we're using sha256 on the token generation pipeline, the possibility of brute forcing the tokens becomes negligible.

### CODE REVIEW

File: src/components/helpers.js [hunk 1]
```diff
@@ -4,6 +4,7 @@ import { Component } from 'preact';
 import { Livechat, useSsl } from '../api';
 import I18n from '../i18n';
 import store from '../store';
```
Comment (murtaza98) [inline src/components/helpers.js:6]: @cauefcr There are a couple of formatting issues introduced here due to the new import. Could you please take care of that. Sidenote: try installing tslint or eslint extension on you vscode - they usually detect these errors and fix them automatically 

image

File: src/components/helpers.js [hunk 2]
```diff
@@ -120,7 +121,7 @@ export const setCookies = (rid, token) => {
 	document.cookie = `rc_room_type=l; path=/; ${ getSecureCookieSettings() }`;
 };
 
-export const createToken = () => Math.random().toString(36).substring(2, 15) + Math.random().toString(36).substring(2, 15);
+export const createToken = () => crypto.randomBytes(128).toString('hex');
```
Comment (murtaza98) [inline src/components/helpers.js:124]: Apart from this, I noticed that there are a couple of places elsewhere in the code where we're using this same logic to generate a token :/ Can we please modify that as well to cover all the cases

image

File: src/components/helpers.js (hunk 1 above)
Comment (cauefcr) [reply src/components/helpers.js:6]: Sure, i'm working on it now.

File: src/components/helpers.js (hunk 2 above)
Comment (cauefcr) [reply src/components/helpers.js:124]: great point Murtaza, doing it now.

**Comment by KevLehman** [comment]
@cauefcr , did you try to regenerate loki files? (with `npm run update-storybook`)

**Comment by cauefcr** [comment]
>@cauefcr , did you try to regenerate loki files? (with npm run update-storybook)

yes, the code change is making the components not appear on the storybook, i'm trying to figure out what is wrong

**Comment by KevLehman** [comment]
Hey Caue, did you sign the CLA on this repo? 👀

**Comment by cauefcr** [comment]
> Hey Caue, did you sign the CLA on this repo? eyes

Yes, twice already.

**Review by KevLehman — CHANGES_REQUESTED** [review]

File: src/components/helpers.js [hunk 3]
```diff
@@ -120,7 +121,7 @@ export const setCookies = (rid, token) => {
 	document.cookie = `rc_room_type=l; path=/; ${ getSecureCookieSettings() }`;
 };
 
-export const createToken = () => Math.random().toString(36).substring(2, 15) + Math.random().toString(36).substring(2, 15);
```
[the remark this answers is not in the document]
Comment (cauefcr) [reply src/components/helpers.js:123]: Yes, that's better.

File: src/store/index.js [hunk 4]
```diff
@@ -1,8 +1,9 @@
+import crypto from 'crypto-js';
 import { h, Component, createContext } from 'preact';
 
 import Store from './Store';
 
-const createToken = () => Math.random().toString(36).substring(2, 15) + Math.random().toString(36).substring(2, 15);
+const createToken = () => crypto.lib.WordArray.random(128).toString(crypto.enc.Hex);
```
[the remark this answers is not in the document]
Comment (cauefcr) [reply src/store/index.js:6]: Indeed, too much paranoia, making it smaller.

**Review by KevLehman — APPROVED** [review]

**Pull request merged**